packages/api-fetch/src/index.js

```diff
@@ -38,106 +38,115 @@ const DEFAULT_OPTIONS = {
+	if ( data ) {
+		body = JSON.stringify( data );
+		headers[ 'Content-Type' ] = 'application/json';
+	}
+
+	const responsePromise = window.fetch(
+		url || path,
+		{
+			...DEFAULT_OPTIONS,
+			...remainingOptions,
+			body,
+			headers,
+		}
+	);
+	const checkStatus = ( response ) => {
+		if ( response.status >= 200 && response.status < 300 ) {
+			return response;
+		}
 
-		// Merge explicitly-provided headers with default values.
-		headers = { ...DEFAULT_HEADERS, ...headers };
+		throw response;
+	};
 
-		// The `data` property is a shorthand for sending a JSON body.
-		if ( data ) {
-			body = JSON.stringify( data );
-			headers[ 'Content-Type' ] = 'application/json';
+	const parseResponse = ( response ) => {
+		if ( parse ) {
+			if ( response.status === 204 ) {
+				return null;
+			}
+
+			return response.json ? response.json() : Promise.reject( response );
 		}
 
-		const responsePromise = window.fetch(
-			url || path,
-			{
-				...DEFAULT_OPTIONS,
-				...remainingOptions,
-				body,
-				headers,
-			}
-		);
-		const checkStatus = ( response ) => {
-			if ( response.status >= 200 && response.status < 300 ) {
-				return response;
-			}
+		return response;
+	};
 
-			throw response;
-		};
+	return responsePromise
+		.then( checkStatus )
+		.then( parseResponse )
+		.catch( ( response ) => {
+			if ( ! parse ) {
+				throw response;
+			}
 
-		const parseResponse = ( response ) => {
-			if ( parse ) {
-				if ( response.status === 204 ) {
-					return null;
-				}
+			const invalidJsonError = {
+				code: 'invalid_json',
+				message: __( 'The response is not a valid JSON response.' ),
+			};
 
-				return response.json ? response.json() : Promise.reject( response );
+			if ( ! response || ! response.json ) {
+				throw invalidJsonError;
 			}
 
-			return response;
-		};
-
-		return responsePromise
-			.then( checkStatus )
-			.then( parseResponse )
-			.catch( ( response ) => {
-				if ( ! parse ) {
-					throw response;
-				}
-
-				const invalidJsonError = {
-					code: 'invalid_json',
-					message: __( 'The response is not a valid JSON response.' ),
-				};
-
-				if ( ! response || ! response.json ) {
+			return response.json()
+				.catch( () => {
 					throw invalidJsonError;
-				}
-
-				return response.json()
-					.catch( () => {
-						throw invalidJsonError;
-					} )
-					.then( ( error ) => {
-						const unknownError = {
-							code: 'unknown_error',
-							message: __( 'An unknown error occurred.' ),
-						};
-
-						throw error || unknownError;
-					} );
-			} );
-	};
+				} )
+				.then( ( error ) => {
+					const unknownError = {
+						code: 'unknown_error',
+						message: __( 'An unknown error occurred.' ),
+					};
+
+					throw error || unknownError;
+				} );
+		} );
+};
+
+let fetchHandler = defaultFetchHandler;
+
+function setFetchHandler( newFetchHandler ) {
+	fetchHandler = newFetchHandler;
+}
+
+function apiFetch( options ) {
+	const steps = [ ...middlewares, fetchHandler ];
```

Comment:
Thanks for updating this code here. It is now clear that `fetchHandler` is something unrelated to middlewares.